From 06e6186f2221b3582cd48f1aa421bae0ecb3a581 Mon Sep 17 00:00:00 2001
From: Erik Bengtsson <erik.bengtsson@flir.se>
Date: Wed, 11 Sep 2019 09:00:55 +0200
Subject: [PATCH] lc709203f battery: Only set battery param if not correct. Set
 soc parameters to remove adjustment until curve is decided.

Change of battery param causes initialization.
---
 arch/arm/boot/dts/imx7ulp-ec201.dtsi     |  4 ++--
 drivers/power/supply/lc709203f_battery.c | 15 ++++++++++-----
 2 files changed, 12 insertions(+), 7 deletions(-)

diff --git a/arch/arm/boot/dts/imx7ulp-ec201.dtsi b/arch/arm/boot/dts/imx7ulp-ec201.dtsi
index 55faa48..76f66e3 100644
--- a/arch/arm/boot/dts/imx7ulp-ec201.dtsi
+++ b/arch/arm/boot/dts/imx7ulp-ec201.dtsi
@@ -380,8 +380,8 @@
 		compatible = "onsemi,lc709203f";
 		reg = <0x0b>;
 		thermistor-beta = <0x0d34>;
-		kernel-threshold-soc = <5>;
-		kernel-maximum-soc = <99>;
+		kernel-threshold-soc = <0>;
+		kernel-maximum-soc = <100>;
 		alert-low-rsoc = <0x0>;
 		alert-low-voltage = <0x0>;
 		battery-param = <1>;
diff --git a/drivers/power/supply/lc709203f_battery.c b/drivers/power/supply/lc709203f_battery.c
index 60fbe95..2549305 100644
--- a/drivers/power/supply/lc709203f_battery.c
+++ b/drivers/power/supply/lc709203f_battery.c
@@ -559,11 +559,16 @@ static int lc709203f_probe(struct i2c_client *client,
 	}
 
     if (chip->pdata->battery_param) {
-        ret = lc709203f_write_word(chip->client, LC709203F_CHANGE_OF_THE_PARAM, chip->pdata->battery_param);
-        if (ret < 0) {
-            dev_err(&client->dev, "STATUS_BIT write failed: %d\n", ret);
-            return ret;
-        }
+		u32 param = lc709203f_read_word(chip->client, LC709203F_CHANGE_OF_THE_PARAM);
+
+		if(param != chip->pdata->battery_param)
+		{
+			ret = lc709203f_write_word(chip->client, LC709203F_CHANGE_OF_THE_PARAM, chip->pdata->battery_param);
+			if (ret < 0) {
+				dev_err(&client->dev, "STATUS_BIT write failed: %d\n", ret);
+				return ret;
+			}
+		}
     }
 
 skip_thermistor_config:
-- 
2.7.4

